### PR TITLE
TASK: Add text type to support newer ElasticSearch versions

### PR DESCRIPTION
**Pull request by johannessteu**

### CODE REVIEW

**Comment by johannessteu** [comment]
Regarding breaking, yeah. The change within the EntityMapper would break older installations. That's why i did this PR against master. 

Is the text the best default? Text is the analyzed string, keyword's aren't analyzed. So the old string mapping type should also be analyzed by default as long as you added a `index: not_analyzed`. I'd say text is the correct default as keyword is more specific and something you want to configure yourself 
https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html

**Comment by johannessteu** [comment]
Sooooo ... ?

File: Classes/Indexer/Object/Transform/TextCastTransformer.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * @Flow\Scope("singleton")
+ */
+class TextCastTransformer implements TransformerInterface
+{
+    /**
+     * Returns the Elasticsearch type this transform() method returns
+     *
+     * @return string
+     */
+    public function getTargetMappingType()
```
Comment (daniellienert) [inline Classes/Indexer/Object/Transform/TextCastTransformer.php]:
```suggestion
    public function getTargetMappingType(): string
```

File: Classes/Indexer/Object/Transform/TextCastTransformer.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        return 'text';
+    }
+
+    /**
+     * @param mixed $source
+     * @param TransformAnnotation $annotation
+     * @return string
+     */
+    public function transformByAnnotation($source, TransformAnnotation $annotation)
```
Comment (daniellienert) [inline Classes/Indexer/Object/Transform/TextCastTransformer.php]:
```suggestion
    public function transformByAnnotation($source, TransformAnnotation $annotation): string
```

**Comment by johannessteu** [comment]
Thank's for the suggestion @daniellienert !

**Pull request merged**